lib/compress.js

```diff
@@ -564,25 +564,46 @@ merge(Compressor.prototype, {
             return value && (value.is_constant() || value instanceof AST_Lambda);
         }
 
-        function is_modified(node, level, immutable) {
+        function read_property(obj, key) {
+            if (key instanceof AST_Constant) key = key.getValue();
```

Comment:
Will this work with `AST_RegExp`? It's the old "`AST_Constant` is not constant" problem.

Reply:
Scratch that - this is fine. I was thinking computed properties on harmony.

Reply:
Indeed - this is a check for `AST_Sub.property` being a constant, as I thought forcing it to be only `AST_Dot` seems to limiting.